lib/flop.ex

```diff
@@ -279,8 +388,91 @@ defmodule Flop do
   to render the pagination links anyway, so this shouldn't be a problem.
   """
   @doc since: "0.6.0"
-  @spec meta(Queryable.t(), Flop.t(), keyword) :: Meta.t()
-  def meta(q, flop, opts \\ []) do
+  @spec meta(Queryable.t() | [Schema.t()], Flop.t(), keyword) :: Meta.t()
+  def meta(query_or_results, flop, opts \\ [])
+
+  def meta(
+        results,
+        %Flop{
+          first: first,
+          order_by: order_by,
+          after: nil,
+          before: nil,
+          last: nil,
+          limit: nil
+        } = flop,
+        _opts
+      )
+      when is_list(results) and is_integer(first) do
+    {start_cursor, end_cursor} = get_cursors(results, order_by)
+
+    has_next_page? = length(results) == first + 1
+
+    %{
+      @nil_meta
+      | flop: flop,
+        start_cursor: start_cursor,
+        end_cursor: end_cursor,
+        has_next_page?: has_next_page?,
+        has_previous_page?: false
+    }
+  end
+
+  def meta(
+        results,
+        %Flop{
+          first: first,
+          after: after_,
+          order_by: order_by,
+          before: nil,
+          last: nil,
+          limit: nil
+        } = flop,
+        _opts
+      )
+      when is_list(results) and is_integer(first) and is_binary(after_) do
```

Comment:
This function body is nearly identical to the previous one. Perhaps you can use a single function body and set `has_previous_page?` to `!is_nil(after_)` instead.